### DISCUSSION TITLE
Yamaha MusicCast support

### DISCUSSION THREAD

**Discussion by tomlut** [music-assistant/support#409] [Feature requests and ideas]
Would it be possible to add support for Yamaha MusicCast?

**Comment by erkr** [comment]
I moved this request to the discussions, thanks for filing.

**Comment by HA-TB303** [comment]
I second that. I have several Yamaha MusicCast devices in my house and none of those work a.t.m.

If I add a musiccast mediaplayer the entire integrations stops working (no mediaplayers available, also none of the ones that did work before, like Google Home

**Reply by erkr** [reply]
The title of this issue clearly mentions a support request, Not a bug. I see you filed a new issue, thanks, that one is filed as a Bug though ;-)

**Comment by marcelveldt** [comment]
OK so the bug in the HA integration for MusicCast has been fixed and released in HA 2022.6.7 yesterday: https://github.com/home-assistant/core/pull/73694

Can you guys confirm that MA no longer crashes on MusicCast players and they're now selectable within the MA interface ?
Next step is getting playback working. Let me hear if it works for you and if not, what errors you see etc.

**Comment by princemurphy** [comment]
When trying to play content to my Yamaha receiver, the receiver changes to the server input but displays 'No Content'. My home assistant log says 'Abort: client(s) did not connect within 10 seconds.

**Reply by atomic10** [reply]
Having the same issue princemurphy but noticed it was doing the same when I tried to play from MA to Volumio.
Are you running Home Assistant in a docker container or are you using HAOS?

**Comment by Kuitsi** [comment]
I'm not sure what happened yet other than keeping it to the most recent versions but I managed to get my WXC-50 finally working as it appeared on available players list and playback works for both mp3 and ogg vorbis files.

**Comment by Talw1n** [comment]
I'm looking at getting a Yamaha musiccast vinyl 500 (TTN503) and wanting to use Music Assistant as a middle man to stream it to my speakers via wifi.

Does anyone know if you can play audio from Musiccast devices (amps/turntables) to Music Assistant for it to play it to your speakers (ie. google hub, soundbars, etc.)

I'm thinking I'll need to use the Musiccast integration to get the audio from the deck to then feed it into Music Assistant to then push it out to the speakers.... (at least I'm hoping this may be what is possible)

**Reply by OzGav** [reply]
You can see what  music  providers are supported in the FAQs and MusicCast is not one of them. Also you can see what players providers are supported. Note that in the future HA players will be available to MA but they won’t support all of the MA features.

**Reply by darthdurden** [reply]
@OzGav is there any plan for adding musiccast support in the future?

Really I'd just want to be able to add home assistant media players in and then use the join/unjoin funtionality under the hood, so it doesn't even need "musiccast" support specifically. As of right now in Home Assistant I can only use my players as DLNA or AirPlay which doesn't work with joining.

Is there future support planned for this?

**Reply by OzGav** [reply]
The join and Unjoin functionality works now for players that support that. The exposing of HA entities is high on the to do list but I don’t have an exact time when that will be finished. Whether the exposed HA entities will support joining I am not sure at the moment.

**Reply by darthdurden** [reply]
Thanks for the update! That would be awesome if it did. Musiccast works pretty seamlessly with the join/unjoin stuff, so 🤞 that will just work!

**Reply by darthdurden** [reply]
Oh, is there somewhere I can track the progress of the HA entities? I'd love to follow along!

**Reply by OzGav** [reply]
https://github.com/orgs/music-assistant/projects/2

**Reply by darthdurden** [reply]
Thank you!

**Reply by darthdurden** [reply]
So I finally revisted this. It looks like now I am able to add my musiccast players in as Home Assistant media players. They playback as expected. It doesn't look like the join/unjoin functionality is working though. The players are not linked at all and playback out of sync. Is that expected? Should join/unjoin be working with Home Assistant players?

**Reply by OzGav** [reply]
No it does not as these player types can only be used with universal groups. This may change in the future for HA media player integrations that support perfect sync.

**Reply by darthdurden** [reply]
Bah okay that would be a really nice upgrade and a good way around needing platform specific integrations. Are there any plans for a Musiccast specific integration in lieu of that?

**Reply by OzGav** [reply]
That is what this thread is about. It has 22 votes putting it in the top 20. Of course if a dev steps up to do the work it could happen quickly.

**Comment by BobMac57** [comment]
I vote for MusicCast as well!

**Comment by hcoohb** [comment]
It would be nice indeed.
Currently, when trying to play on the yamaha speakers imported through HA, spotify does not play at all...

**Reply by OzGav** [reply]
If it is only Spotify and other providers play then maybe its not a problem with the Yamaha integration...?

**Comment by Jerrk** [comment]
With the musiccast integration in HA i am able to use it as a HA mediaPlayer in MA and i am able to control playback from MA.

unfortunately this doesn't seem to forward any media info so my receiver just thinks its playing something called "HomeAssistant".

**Comment by nupulu** [comment]
I would also like to see native support for Yamaha MusicCast Player Provider.

It is already possible to integrate MusicCast players into HA and use the Join/Unjoin actions there to create groups and then use MA to play to such a group through the HA player provider.

Also each individual MusicCast player can be used in MA through the HA player provider, but grouping in MA is only possible the Universal Group type with the drawback of not supporting perfect sync.

It would be more flexible and convenient if MA were able to use MusicCast players directly /natively without using the HA play provider. This would allow for grouping MusicCast players natively in MA with perfect sync.

Note: the MusicCast devices also support AirPlay and can be grouped natively in MA via the AirPlay player provider. However, when using a MusicCast Amplivier/Reciever it is better to provide all devices through the MultiCast protocol/API to avoid a fiddling back and forth as the devices often get into a "stuck" state where you need to reboot them. 

Another worthwile addition would be to have MusicCase as a Music Provider as MusicCast Receivers are able to provide audio streams from any of their inputs, i.e. not only the streaming providers such as Tidal etc, but more interestingly, also the physical inputs, such as Audio 1...n, HDMI 1..m. I don't know how much sense this makes, as this would be just audio streams, not content that has individual track names, only the name of the input (and for HDMI its possible CEC tag) would be available.

**Comment by frank2604** [comment]
Me too!

I have the hole multiroom setup in my house on MusicCast with AV receiver, soundbars, subs and a lot of WX-021. At the moment I run MA via the Airplay provider. After playing around with the Airplay audio buffer it seems to be stable with 1.500 ms.

I have also figured out, that media_player.turn_off action on the **MA entity** in HA stops the playback and in the MA UI the player is shown as OFF, but it is not really OFF. The LED indicator on the device is ON and the **MusicCast entity** in HA is ON too. On the other hand if you use the **MusicCast entity** to turn off, MA do not recognize this and then everything is broken.

**Comment by tmuir1** [comment]
I too would like Musiccast to be able to have more native functionality within MA. Would be great if it could be used as a player provider just like Sonos...

**Comment by kmidt** [comment]
I Wish native musiccast Support too

**Comment by BobMac57** [comment]
Lack of support for Musiccast is what drove me away from MA...when it is implemented, I will come back.

**Comment by gsemet** [comment]
Same, would love a native integration, and it sees the protocol is not so encrypted and there are interesting Python libraries already.

**Comment by BobMac57** [comment]
This request has been entered in 2022 and nothing from the devs... I'm seriously thinking this will never happen.

**Reply by marcelveldt** [reply]
Why do you think that? All it takes is a volunteer that wants to create a provider or enough votes so one of the core members of the project will pick it up.

**Reply by BobMac57** [reply]
I know the process but, it's been 3 years and no one has stepped up.

**Reply by Jerrk** [reply]
if it's that important to you, you could've learned python in those 3 years and fixed it yourself.

**Reply by BobMac57** [reply]
Assh*le... living up to that name I see....

**Reply by Kernald** [reply]
Can we maybe:
1) Keep it civil
2) Just abstain from commenting and not adding anything to the actual feature?
Thanks.

**Reply by OzGav** [reply]
See the above. Any further commentary not complying with that will result in a ban.

**Comment by OzGav** [comment]
MusicCast support has been added in 2.6.0b6

**Comment by chris-hemmings** [comment]
Just set it up on the beta, and loving it so far. Seamless. Great work!

**Comment by dmutsaers** [comment]
I’m excited to see MusicCast support introduced in the Music Assistant beta—great work! That said, I’ve run into a few issues related to volume control on MusicCast media players provided by Music Assistant.

When `volume_step: 1` is enabled in the Mini Media Player card, increasing the volume doesn’t work at all. If I remove that setting, volume control becomes functional again, but the adjustment is quite large—jumping by +5 instead of a more gradual increase.

I’ve also noticed a mismatch in volume reporting between the MusicCast media player provided by Music Assistant and the Yamaha (YNCA) media player. The Yamaha (YNCA) player seems to reflect volume levels accurately, while the MusicCast player provided by Music Assistant does not.
<img width="1546" height="355" alt="image" src="https://github.com/user-attachments/assets/612faea8-6718-4ab6-99a6-3a06a0918b9a" />

Top: Yamaha (YNCA) integration
Bottom: Music Assistant MusicCast integration

**Reply by OzGav** [reply]
@fmunkes

**Reply by fmunkes** [reply]
For the volume reporting - you are correct, I forgot to subtract the min volume when converting to percentage, which might be what you see. For this specific device, is the minimal volume not zero? What kind of player do you have? Do you have another player you can verify with - for example, for a MusicCast 20 everything is fine and in sync with the MusicCast app. I'll create a PR for the volume reporting, but would like to include other changes if needed...

**Reply by dmutsaers** [reply]
Hi, Thanks for looking into this.

I'm seeing a discrepancy in volume reporting on my CX-A5100, which uses dB-based volume levels (with -80 dB as the minimum). The other devices involved are the XDA-QS5400 and the MusicCast 20.

On the MusicCast 20, the Music Assistant MusicCast provider reports the volume as 50%, while the official MusicCast app shows it as 30%. This mismatch seems consistent across devices.

Additionally, I'm unable to increase the volume on any Music Assistant MusicCast device when volume_step: 1 and/or volume_stateless: true are set. This appears to block volume increments entirely..

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
